lib/trino-filesystem/src/main/java/io/trino/filesystem/TrinoFileSystem.java

```diff
@@ -153,6 +153,27 @@ void renameFile(Location source, Location target)
     FileIterator listFiles(Location location)
             throws IOException;
 
+    /**
+     * Lists all files within the specified directory recursively or non-recursively .Default
+     * implementation is recursively. The location can be empty,
+     * listing all files in the file system, otherwise the location must end with a slash. If the
+     * location does not exist, an empty iterator is returned.
+     * <p>
+     * For hierarchical file systems, if the path is not a directory, an exception is
+     * raised.
+     * For hierarchical file systems, if the path does not reference an existing
+     * directory, an empty iterator is returned. For blob file systems, all blobs
+     * that start with the location are listed. In the rare case that a blob exists with the
+     * exact name of the prefix, it is not included in the results.
+     * <p>
+     * The returned FileEntry locations will start with the specified location exactly.
+     *
+     * @param location the directory to list
+     * @throws IllegalArgumentException if location is not valid for this file system
+     */
+    FileIterator listFiles(Location location, boolean isRecursive)
```

Comment:
If we need recursive and non-recursive file listings, i would consider naming the methods differently, instead of a boolean flag.
note that 
`listFiles` is recursive 
`listDirectories` is non recursive

that might be confusing.

besides hudi, the other place where non-recursive file listing would be very helpful is FileHiveMetastore (https://github.com/trinodb/trino/issues/20130)

anyway, i assume David might be interested in reviewing the changes here
as well as Elon and Dain, who wrote some filesystem implementations
what about extracting the filesystem changes to a separate PR?

Reply:
@findepi if I create separate PR then I need raised this PR after listing recursive one is merged .

Reply:
@findepi   I will rename the method name